st2client/st2client/commands/trace.py

```diff
@@ -299,7 +304,25 @@ def _apply_display_filters(trace, args):
         should be displayed.
         """
         # If all the filters are false nothing is to be filtered.
-        if not(args.show_executions or args.show_rules or args.show_trigger_instances):
+        all_component_types = not(args.show_executions or
+                                  args.show_rules or
+                                  args.show_trigger_instances)
+
+        # check if noop_triggers are to be hidden. This check applies whenever TriggerInstances
+        # are to be shown.
+        if (all_component_types or args.show_trigger_instances) and args.hide_noop_triggers:
+            filtered_trigger_instances = []
+            for trigger_instance in trace.trigger_instances:
+                is_noop_trigger_instance = True
+                for rule in trace.rules:
+                    caused_by_id = rule.get('caused_by', {}).get('id', None)
+                    if caused_by_id == trigger_instance['object_id']:
```

Comment:
Yeah it is terrible. Given that we would not be dealing with rather large collections here I think it is an ok option.